### PR TITLE
closes #4 - Add phpcs-security-audit to our phpcs configuration

### LINKED ISSUE #4: Add phpcs-security-audit to our phpcs configuration
OWASP.org recommends using source code analysis tools for static application security analysis and testing part of a build process / continuous integration.

We should look into adding the following tool (listed on [OWASP's site](https://owasp.org/www-community/Source_Code_Analysis_Tools)) into our existing phpcs configuration:
https://github.com/FloeDesignTechnologies/phpcs-security-audit

### PR DESCRIPTION
**Pull request by kevcooper**
This adds pheromone/phpcs-security-audit to the dev dependencies.

Related to this PR, do we also want to include sensiolabs/security-checker as another dependency or rely on the builtin functionality within Symfony? (`symfony security:check`)

This is a blocker for working on static code analysis CI pipelines unless we'd prefer to bake these tools into a docker image instead of listing them as dev dependencies.

### CODE REVIEW

**Comment by joeparsons** [comment]
I think we'll also need to add some phpcs-security-audit specific rules to our existing [phpcs.xml.dist](https://github.com/az-digital/az-quickstart-scaffolding/blob/master/phpcs.xml.dist) file for this to do anything.

**Comment by joeparsons** [comment]
Acquia's coding-standards-php project makes use of phpcs-security-audit and contains some example phpcs config files that make use of its rules that we could copy from.  We may also want to consider adding their coding-standards-php project as a dependency to our project or consider creating our own quickstart-coding-standards-php composer package that could be used by our other repos.
https://github.com/acquia/coding-standards-php

**Comment by tadean** [comment]
This looks promising! The default **az-quickstart** standard now is showing as including the **PHPCS_SecurityAudit** sniffs for me:

./vendor/bin/phpcs -e
```
...
PHPCS_SecurityAudit (20 sniffs)
-------------------------------
  PHPCS_SecurityAudit.BadFunctions.Asserts
  PHPCS_SecurityAudit.BadFunctions.Backticks
  PHPCS_SecurityAudit.BadFunctions.CallbackFunctions
  PHPCS_SecurityAudit.BadFunctions.CryptoFunctions
  PHPCS_SecurityAudit.BadFunctions.EasyRFI
  PHPCS_SecurityAudit.BadFunctions.EasyXSS
  PHPCS_SecurityAudit.BadFunctions.ErrorHandling
  PHPCS_SecurityAudit.BadFunctions.FilesystemFunctions
  PHPCS_SecurityAudit.BadFunctions.FringeFunctions
  PHPCS_SecurityAudit.BadFunctions.FunctionHandlingFunctions
  PHPCS_SecurityAudit.BadFunctions.Mysqli
  PHPCS_SecurityAudit.BadFunctions.NoEvals
  PHPCS_SecurityAudit.BadFunctions.Phpinfos
  PHPCS_SecurityAudit.BadFunctions.PregReplace
  PHPCS_SecurityAudit.BadFunctions.SQLFunctions
  PHPCS_SecurityAudit.BadFunctions.SystemExecFunctions
  PHPCS_SecurityAudit.Drupal8.CVE20132110
  PHPCS_SecurityAudit.Drupal8.CVE20134113
  PHPCS_SecurityAudit.Misc.BadCorsHeader
  PHPCS_SecurityAudit.Misc.IncludeMismatch
```

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,9 @@
     "require-dev": {
         "dealerdirect/phpcodesniffer-composer-installer": "^0.5.0",
         "drupal/core-dev": "8.8.0",
-        "mglaman/drupal-check": "^1.0"
+        "mglaman/drupal-check": "^1.0",
+        "pheromone/phpcs-security-audit": "^2.0",
+        "squizlabs/php_codesniffer": "^3.5"
```
Comment (joeparsons) [inline composer.json]: This is unnecessary because it's already provided by drupal/core-dev

**Comment by kevcooper** [comment]
Looks like the issue with security-checker is that neither of the lock files we're checking have any packages defined in them and the tool doesn't handle this gracefully

**Comment by joeparsons** [comment]
Remaining tasks:
- Change script to only  check main composer.lock file
- Resolve conflicts

File: composer.json [hunk 2]
```diff
@@ -62,6 +64,9 @@
         "post-update-cmd": [
             "@composer drupal:scaffold",
             "QuickstartProject\\composer\\ScriptHandler::createRequiredFiles"
+        ],
+        "sniff": [
+            "phpcs -p --colors --standard=phpcs.xml.dist --extensions=php,install,module,profile,inc"
```
[the remark this answers is not in the document]
Comment (joeparsons) [reply composer.json]: That'd probably be easier, yeah.

**Pull request merged**